perception/autoware_ground_filter/test/test_ground_filter.cpp

```diff
@@ -0,0 +1,322 @@
+// Copyright 2024 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "autoware/ground_filter/ground_filter.hpp"
+
+#include <autoware/point_types/types.hpp>
+
+#include <gtest/gtest.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+#include <pcl_conversions/pcl_conversions.h>
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+class GroundFilterTest : public ::testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    // Initialize parameter structure
+    param_.global_slope_max_angle_rad = 0.26f;  // ~15 degrees
+    param_.local_slope_max_angle_rad = 0.26f;   // ~15 degrees
+    param_.radial_divider_angle_rad = 0.0175f;  // 1 degree
+    param_.use_recheck_ground_cluster = true;
+    param_.use_lowest_point = true;
+    param_.detection_range_z_max = 2.0f;
+    param_.non_ground_height_threshold = 0.2f;
+    param_.grid_size_m = 0.5f;
+    param_.grid_mode_switch_radius = 20.0f;
+    param_.ground_grid_buffer_size = 3;
+    param_.virtual_lidar_x = 1.4f;
+    param_.virtual_lidar_y = 0.0f;
+    param_.virtual_lidar_z = 1.9f;
+
+    // Create filter
+    ground_filter_ = std::make_unique<autoware::ground_filter::GroundFilter>(param_);
+
+    // Create sample point cloud
+    createSamplePointCloud();
+  }
+
+  void TearDown() override { ground_filter_.reset(); }
+
+  void createSamplePointCloud()
+  {
+    auto cloud = std::make_shared<sensor_msgs::msg::PointCloud2>();
+
+    // Create simple XYZ point cloud for testing
+    pcl::PointCloud<pcl::PointXYZ> pcl_cloud;
+
+    // Add ground points
+    for (int i = 0; i < 50; ++i) {
+      pcl::PointXYZ point;
+      point.x = static_cast<float>(i % 10) - 5.0f;
+      point.y = static_cast<float>(i / 10) - 2.5f;
```

Comment:
Before conversion into float, the divided value becomes zero. For example, when i = 1 ~ 9, the value does not become 0.1, ... 0.9, but it becomes zero.
```suggestion
      point.y = static_cast<float>(i / 10.) - 2.5f;
```

You can see the behavior by using the following toy code.
```
#include <iostream>

int main() {
    int i = 1;
    float f = static_cast<float>(i / 10) - 2.5f;
    std::cout << f << std::endl;
}
```

Reply:
Ah, sorry maybe this is my misunderstanding. This for loop is inserting the point as grid-like. Let me once close my other same proposals.